### DISCUSSION TITLE
Links stopping short. Need to travel longer based on link width

### DISCUSSION THREAD

**Discussion by JeremyGEA** [Blazor-Diagrams/Blazor.Diagrams#251] [Q&A] [answered]
I've noticed some things on the line routers that I see as an issue. We normally use Orthogonal Routing but the issue persists with with Bezier (curved) lines as well. The links stop a bit early. They do not continue to the node but rather stop at the outward most quadrant of the connected port. When the line widths are larger than a few pixels, this causes a very strange look. Can we have the link continue to the node-edge itself? Perhaps this could be fixed by just making the ports flush with the node-edge but perhaps there is another way or option?

Another slight issue is with the Orthogonal router. When it makes a 90 degree bend, it stops each line right at the point of the bend. It really needs to continue each line for another link width / 2 in order to not leave a gap and make a nice sharp corner. I've looked into making this adjustment and I'm having trouble following the orthogonal router code. Could someone help me here?

Here are some examples with enlarged line widths. We aren't using this large of a line but you can see the same effect at just 5-15px width.
Orthogonal Links (see the line ends and at each bend):
![image](https://user-images.githubusercontent.com/49650352/194066563-4f563a18-2469-4681-8429-1ba602de76fb.png)
Bezier Links:
![image](https://user-images.githubusercontent.com/49650352/194066829-d37c1073-3d0a-4ac6-813c-844b5cb3f3aa.png)

**Comment by zHaytam** [comment, marked as answer]
Hello,

That's because SinglePortAnchor chooses the position based on the anchor's shape by default. Try setting `MiddleIfNoMarker` to `true` when creating links (factory or manual).

**Reply by JeremyGEA** [reply]
Correct you are. MiddleIfNoMarker fixed the links at the node connection.
![image](https://user-images.githubusercontent.com/49650352/194371036-ff4fb68b-6a92-4b6c-a2f5-1c2292fb28be.png)

Any thoughts on the Orthogonal bends?

**Reply by zHaytam** [reply]
Since you're not using markers, you an use `stroke-linecap: square;` in CSS on links (not selection helpers).

**Reply by JeremyGEA** [reply]
Good idea on the `stroke-linecap`!
I've tried adding `stroke-linecap: square` to `.link` and anywhere else I can think of in CSS and nothing seems to work.
If I add it to the `<path>` of LinkWidget, it works great:
`<path d="@d"
          stroke-linecap="square"
          stroke-width="@Link.Width.ToInvariantString()"
          fill="none"
          stroke="@color"/>`

Where in CSS is this supposed to go or could I make a PR for adding a linecap property to the LinkModel like color and width?
